modules/database/database-sql/src/main/java/org/eclipse/dirigible/database/sql/dialects/hana/HanaSqlDialect.java

```diff
@@ -26,39 +20,51 @@
 import org.eclipse.dirigible.database.sql.builders.records.UpdateBuilder;
 import org.eclipse.dirigible.database.sql.dialects.DefaultSqlDialect;
 
+import java.sql.*;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+
 /**
  * The HANA SQL Dialect.
  */
 public class HanaSqlDialect extends
         DefaultSqlDialect<SelectBuilder, InsertBuilder, UpdateBuilder, DeleteBuilder, HanaCreateBranchingBuilder, AlterBranchingBuilder, DropBranchingBuilder, HanaNextValueSequenceBuilder, HanaLastValueIdentityBuilder>
-        implements DatabaseArtifactTypes{
+        implements DatabaseArtifactTypes {
 
     private static final String IDENTITY_ARGUMENT = "GENERATED BY DEFAULT AS IDENTITY (START WITH 1 INCREMENT BY 1)";
 
     private boolean isSequenceExisting(Connection connection, String sequence) throws SQLException {
-        String sql = "ALTER SEQUENCE \"" + sequence +"\"";
+        String sql = "ALTER SEQUENCE \"" + sequence + "\"";
         PreparedStatement statement = connection.prepareStatement(sql);
         try {
             statement.executeUpdate();
             return true;
-        }catch(Exception e){
-
+        } catch (Exception e) {
             throw new SQLException("Sequence does not exist: " + sequence);
         }
     }
-    private boolean isProcedureExisting(Connection connection, String procedure) throws  SQLException{
+
+    private boolean isProcedureExisting(Connection connection, String procedure) throws SQLException {
         DatabaseMetaData metadata = connection.getMetaData();
-        ResultSet procedureDescription = metadata.getProcedures(null, null,procedure);
-        return procedureDescription.next();
+        ResultSet procedureDescription = metadata.getProcedures(null, null, procedure);
+        if (procedureDescription.next()) {
+            return true;
+        }
+        throw new SQLException("Cannot find procedure: " + procedure);
     }
 
-    private boolean isFunctionExisting(Connection connection, String function) throws SQLException{
+    private boolean isFunctionExisting(Connection connection, String function) throws SQLException {
         DatabaseMetaData metadata = connection.getMetaData();
-        ResultSet funcDescription = metadata.getFunctions(null,null,function);
-        return funcDescription.next();
+        ResultSet funcDescription = metadata.getFunctions(null, null, function);
+        if (funcDescription.next()) {
+            return true;
+        }
+        throw new SQLException("Cannot find function: " + function);
```

Comment:
Should we really throw an exception here.

How about the following instead?
```java
@Override
    public boolean exists(Connection connection, String artefact, int type) throws SQLException {
        try {
            switch(type) {
                case DatabaseArtifactTypes.TABLE:
                case DatabaseArtifactTypes.VIEW:
                case DatabaseArtifactTypes.SYNONYM:
                    count(connection, artefact);
                    break;
                case DatabaseArtifactTypes.FUNCTION:
                    return isFunctionExisting(connection, artefact);
                case DatabaseArtifactTypes.PROCEDURE:
                    return isProcedureExisting(connection, artefact);
                case DatabaseArtifactTypes.SEQUENCE:
                    return isSequenceExisting(connection, artefact);
                default:
                    return false;
            }

        } catch(Exception e) {
            return false;
        }
        return true;
    }
```

Or even better:
```java
@Override
    public boolean exists(Connection connection, String artefact, int type) throws SQLException {
        boolean exists = false;
        try {
            switch(type) {
                case DatabaseArtifactTypes.TABLE:
                case DatabaseArtifactTypes.VIEW:
                case DatabaseArtifactTypes.SYNONYM:
                    exists = count(connection, artefact) > 0;
                    break;
                case DatabaseArtifactTypes.FUNCTION:
                    exists = isFunctionExisting(connection, artefact);
                    break;
                case DatabaseArtifactTypes.PROCEDURE:
                    exists = isProcedureExisting(connection, artefact);
                    break;
                case DatabaseArtifactTypes.SEQUENCE:
                    exists = isSequenceExisting(connection, artefact);
                    break;
            }

        } catch(Exception e) {
            // Do nothing, or log message
        }
        return exists;
    }
```